### PR TITLE
SWC-7143

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

File: packages/synapse-react-client/src/components/EntityBadgeIcons/EntityBadgeIcons.tsx [hunk 1]
```diff
@@ -104,6 +105,11 @@ export const EntityBadgeIcons = (props: EntityBadgeIconsProps) => {
 
   const { ref, inView } = useInView()
 
+  const { data: entityData } = useGetJson(entityId, versionNumber, true, {
```
Comment (jay-hodgson) [inline packages/synapse-react-client/src/components/EntityBadgeIcons/EntityBadgeIcons.tsx:108]: Get the derived annotations!

File: packages/synapse-react-client/src/components/EntityBadgeIcons/EntityBadgeIcons.tsx [hunk 2]
```diff
@@ -104,6 +105,11 @@ export const EntityBadgeIcons = (props: EntityBadgeIconsProps) => {
...
+    enabled: inView,
+    staleTime: 60 * 1000, // 60 seconds
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/EntityBadgeIcons/EntityBadgeIcons.tsx:110]: Consider removing the `staleTime` override here and in the call to `useGetEntityBundle`. I don't think these are serving any purpose
```suggestion
```

File: packages/synapse-react-client/src/components/EntityBadgeIcons/EntityBadgeIcons.tsx [hunk 3]
```diff
@@ -118,7 +124,25 @@ export const EntityBadgeIcons = (props: EntityBadgeIconsProps) => {
...
+  ) => {
+    if (typeof value === 'string') {
+      return value
+    } else if (Array.isArray(value)) {
+      return value.join(', ')
+    } else {
+      // Handle number and boolean cases if needed
+      return value?.toString()
+    }
+  }
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/EntityBadgeIcons/EntityBadgeIcons.tsx:145]: you may be able to reuse / modify `getDisplayedAnnotation` in AnnotationsTable.tsx

**Review by nickgros — APPROVED** [review]
nice! a few suggestions

File: packages/synapse-react-client/src/components/EntityBadgeIcons/EntityBadgeIcons.tsx (hunk 3 above)
Comment (jay-hodgson) [reply packages/synapse-react-client/src/components/EntityBadgeIcons/EntityBadgeIcons.tsx]: Reusing, which improves the rendering of datetimes.  Great idea

**Review by nickgros — APPROVED** [review]

**Pull request merged**